support/cas-server-support-x509-core/src/main/java/org/apereo/cas/adaptors/x509/authentication/principal/DefaultX509AttributeExtractor.java

```diff
@@ -0,0 +1,71 @@
+package org.apereo.cas.adaptors.x509.authentication.principal;
+
+import org.apereo.cas.util.CollectionUtils;
+import org.apereo.cas.util.LoggingUtils;
+
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.apache.commons.lang3.StringUtils;
+import java.security.cert.CertificateParsingException;
+import java.security.cert.X509Certificate;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Get the default set of attributes from an X509 certificate.
+ * @author Hal Deadman
+ * @since 6.4
+ */
+@Slf4j
+public class DefaultX509AttributeExtractor implements X509AttributeExtractor {
+
+    @Override
+    public Map<String, List<Object>> extractPersonAttributes(final X509Certificate certificate) {
+        val attributes = new LinkedHashMap<String, List<Object>>();
+
+        if (certificate != null) {
+            if (StringUtils.isNotBlank(certificate.getSigAlgOID())) {
+                attributes.put("sigAlgOid", CollectionUtils.wrapList(certificate.getSigAlgOID()));
+            }
+            val subjectDn = certificate.getSubjectDN();
+            if (subjectDn != null) {
+                attributes.put("subjectDn", CollectionUtils.wrapList(subjectDn.getName()));
+            }
+            val subjectPrincipal = certificate.getSubjectX500Principal();
+            if (subjectPrincipal != null) {
+                attributes.put("subjectX500Principal", CollectionUtils.wrapList(subjectPrincipal.getName()));
+            }
+            val issuerDn = certificate.getIssuerDN();
+            if (issuerDn != null) {
+                attributes.put("issuerDn", CollectionUtils.wrapList(issuerDn.getName()));
+            }
+            val issuerPrincipal = certificate.getIssuerX500Principal();
+            if (issuerPrincipal != null) {
+                attributes.put("issuerX500Principal", CollectionUtils.wrapList(issuerPrincipal.getName()));
+            }
+            try {
+                val rfc822Email = X509ExtractorUtils.getRFC822EmailAddress(certificate.getSubjectAlternativeNames());
+                if (rfc822Email != null) {
+                    attributes.put("x509Rfc822Email", CollectionUtils.wrapList(rfc822Email));
+                }
+            } catch (final CertificateParsingException e) {
+                if (LOGGER.isDebugEnabled()) {
```

Comment:
This is usually not a good idea and sets an unfortunate precedent. If the idea is to not log the trace if debug is off, that should be handled within the Util class. Otherwise, as before, this only decreases the test coverage.